### DISCUSSION TITLE
Error when publish to IIS

### DISCUSSION THREAD

**Discussion by mig0000** [kubernetes-client/csharp#1031] [Q&A]
Hi, 
I did the application with .Net Core 6 and I'm trying to publish it on a server with IIS. The application work ok, but can't connecto to Kubernetes (kubectl is working correctly).

That´s the error:

System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> System.Security.Authentication.AuthenticationException: Authentication failed because the remote party sent a TLS alert: 'HandshakeFailure'.

I try to use a impersonate with my user, use the local kubeconfig, put a kubeconfig file for the connection, but i can't make it work

**Comment by tg123** [comment]
what is your server os?
seems server rejected the tls handshake, probably because tls version too old

**Reply by mig0000** [reply]
Windows Server 2022 Standard

**Reply by tg123** [reply]
can you confirm if dotnet standalone has same issue?

**Reply by mig0000** [reply]
the some

**Reply by tg123** [reply]
could you please also double confirm your k8s server ver?
this err is due to tls version is not supported by os

see full list here https://learn.microsoft.com/en-au/windows/win32/secauthn/cipher-suites-in-schannel

**Reply by mig0000** [reply]
Yes, it is Azure AKS server. I'm connecting and working from my laptop. The problem is publish the application on preproducton server...

It´s the kubernetes versión: 1.22.6

**Reply by tg123** [reply]
could you please test
ServicePointManager.SecurityProtocol = Tls12;

https://learn.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.securityprotocol?view=net-6.0

I will pick a win2022 to test to see what happens

**Reply by mig0000** [reply]
Very thanks for your help. I try to put that property but its the some.

I think its a rule from IT (domain policies or any like this). I create a local VM machine with the some operating system and there work correctly. :P

**Reply by tg123** [reply]
![image](https://user-images.githubusercontent.com/170430/192911300-b5034dfc-cd16-4e67-841a-30ae9b7b995c.png)

could you please check with browser (edge builtin 2022)
I tested connecting to AKS on server 2022, works well for me
